### PR TITLE
Improve name and description of File Downloader Algorithm and fix failing SKIP OUTPUT

### PR DESCRIPTION
**Pull request by signedav**
- Changed name to "Download file (HTTP request)" because it's not given that a file download is performed by HTTP, and this is what this algorithm is about. Even more one easily can use it to perform HTTP requests without caring about the file.
- Still the algorithm failed because of SKIP OUTPUT not passing a file path to the QgsFileDownloader why I set the parameter to mandatory instead
- Additionally I added some tags

One could think about other improvements (but not part of this PR):
- Allow not to download files and only perform requests
- Put responses and / or status code (even on error) to the output

### CODE REVIEW

**Comment by nyalldawson** [comment]
> Allow not to download files and only perform requests

I'd lean toward handling non-download requests as a separate algorithm (or algorithms?) which is specialised for that task only. Eg a "POST data" algorithm, with options specific to POST requests only. That's more inline with the processing philosophy of smaller, specialised tools with options specific to one task, vs the GRASS approach of tools which cover a whole range of uses and with many corresponding options.

**Comment by signedav** [comment]
> I'd lean toward handling non-download requests as a separate algorithm (or algorithms?) which is specialised for that task only. Eg a "POST data" algorithm, with options specific to POST requests only. That's more inline with the processing philosophy of smaller, specialised tools with options specific to one task, vs the GRASS approach of tools which cover a whole range of uses and with many corresponding options.

Yes. This sounds reasonable.

So I keep this PR to provide a more specific name, and in case I would implement a non-download option I would create a new algorithm.

**Comment by nyalldawson** [comment]
@signedav sounds good. I would lean towards renaming to "Download file via HTTP(S)"

**Comment by signedav** [comment]
I agree on the name suggestion of @nyalldawson 

Concerned renaming and added a short description.

image

File: src/analysis/processing/qgsalgorithmfiledownloader.cpp [hunk 1]
```diff
@@ -35,12 +35,17 @@ QString QgsFileDownloaderAlgorithm::name() const
 
 QString QgsFileDownloaderAlgorithm::displayName() const
 {
-  return tr( "Download file" );
+  return tr( "Download file via HTTP(S)" );
+}
+
+QString QgsFileDownloaderAlgorithm::shortDescription() const
+{
+  return tr( "Downloads a URL on the file system with an HTTP(S) GET or POST request" );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmfiledownloader.cpp:43]:
```suggestion
  return tr( "Downloads a URL to the file system with an HTTP(S) GET or POST request" );
```

File: src/analysis/processing/qgsalgorithmfiledownloader.cpp [hunk 2]
```diff
@@ -55,7 +60,7 @@ QString QgsFileDownloaderAlgorithm::groupId() const
 
 QString QgsFileDownloaderAlgorithm::shortHelpString() const
 {
-  return tr( "This algorithm downloads a URL on the file system." );
+  return tr( "This algorithm downloads a URL on the file system with an HTTP(S) GET or POST request" );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmfiledownloader.cpp:63]:
```suggestion
  return tr( "This algorithm downloads a URL to the file system with an HTTP(S) GET or POST request" );
```

**Review by nyalldawson — APPROVED** [review]

File: src/analysis/processing/qgsalgorithmfiledownloader.cpp (hunk 2 above)
Comment (signedav) [reply src/analysis/processing/qgsalgorithmfiledownloader.cpp]: Thanks

**Pull request merged**